src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -343,18 +343,31 @@ private async Task SolutionManager_CacheUpdatedAsync(TimeSpan timeSpan, string e
 
         private async ValueTask RefreshWhenNotExecutingActionAsync(RefreshOperationSource source, TimeSpan timeSpanSinceLastRefresh)
         {
-            var sw = Stopwatch.StartNew();
             // Only refresh if there is no executing action. Tell the operation execution to refresh when done otherwise.
             if (_isExecutingAction)
             {
                 _isRefreshRequired = true;
-                EmitRefreshEvent(timeSpanSinceLastRefresh, source, RefreshOperationStatus.NoOp, isUIFiltering: false, duration: 0);
+                EmitRefreshEvent(timeSpanSinceLastRefresh, source, RefreshOperationStatus.NoOp, isUIFiltering: false);
             }
             else
             {
-                await RefreshAsync();
-                sw.Stop();
-                EmitRefreshEvent(timeSpanSinceLastRefresh, source, RefreshOperationStatus.Success, isUIFiltering: false, duration: sw.Elapsed.TotalMilliseconds);
+                var refreshStatus = RefreshOperationStatus.NoOp;
+                var sw = Stopwatch.StartNew();
+                try
+                {
+                    await RefreshAsync();
```

Comment:
Does RefreshAsync() do just call to IO to get data or do both IO and repopulating the the UI?

Reply:
It does both.